packages/core/src/services/documentLogs/_createDocumentLogQuery.ts

```diff
@@ -0,0 +1,52 @@
+import { eq, isNotNull, or, sum } from 'drizzle-orm'
+
+import { Commit } from '../../browser'
+import { database } from '../../client'
+import { DocumentLogsRepository } from '../../repositories/documentLogsRepository'
+import { commits, providerLogs } from '../../schema'
+
+export function createDocumentLogQuery(workspaceId: number, db = database) {
+  const documentLogsScope = new DocumentLogsRepository(workspaceId, db)
+  const scope = documentLogsScope.scope
+
+  const aggregatedFieldsSubQuery = db
+    .select({
+      id: scope.id,
+      tokens: sum(providerLogs.tokens).mapWith(Number).as('tokens'),
+      durationInMs: sum(providerLogs.duration)
+        .mapWith(Number)
+        .as('duration_in_ms'),
+      costInMillicents: sum(providerLogs.cost_in_millicents)
+        .mapWith(Number)
+        .as('cost_in_millicents'),
+    })
+    .from(scope)
+    .innerJoin(providerLogs, eq(providerLogs.documentLogUuid, scope.uuid))
+    .groupBy(scope.id)
+    .as('aggregatedFieldsSubQuery')
+
+  return {
+    scope,
+    aggregatedFieldsSubQuery,
+    baseQuery: db
+      .select({
+        ...scope._.selectedFields,
+        commit: commits,
+        tokens: aggregatedFieldsSubQuery.tokens,
+        durationInMs: aggregatedFieldsSubQuery.durationInMs,
+        costInMillicents: aggregatedFieldsSubQuery.costInMillicents,
+      })
+      .from(scope)
+      .innerJoin(commits, eq(commits.id, scope.commitId))
+      .innerJoin(
+        aggregatedFieldsSubQuery,
+        eq(aggregatedFieldsSubQuery.id, scope.id),
+      ),
+  }
+}
+
+export function getCommitFilter(draft?: Commit) {
+  return draft
+    ? or(isNotNull(commits.mergedAt), eq(commits.id, draft.id))
+    : isNotNull(commits.mergedAt)
+}
```

Comment:
nothing fancy, I DRYed some of the code in that long method from DocumentVersionsRepository since I need it in two places now